### PR TITLE
ensure mockability of service client models

### PR DESCRIPTION
**Pull request by patilsnr**

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

**Comment by timtay-microsoft** [comment]
There are other models in the service client (device, module, etc.). Are those classes already done?

**Comment by patilsnr** [comment]
> There are other models in the service client (device, module, etc.). Are those classes already done?

@ngastelum-ms should be handling the rest of the service client models.

File: iothub/service/src/DigitalTwin/Models/DigitalTwinGetResponse.cs [hunk 1]
```diff
@@ -14,9 +14,12 @@ public class DigitalTwinGetResponse<T>
         /// <summary>
         /// Creates an instance of this class.
         /// </summary>
+        /// <remarks>
+        /// This class can be inherited from and set by unit tests for mocking purposes.
+        /// </remarks>
         /// <param name="eTag">Weak ETag of the modified resource.</param>
         /// <param name="digitalTwin">The deserialized digital twin.</param>
-        internal DigitalTwinGetResponse(T digitalTwin, ETag eTag = default)
+        protected internal DigitalTwinGetResponse(T digitalTwin, ETag eTag = default)
```
Comment (drwill-ms) [inline iothub/service/src/DigitalTwin/Models/DigitalTwinGetResponse.cs:22]: A step further would be to make the properties that are initialized via a ctor as readonly (meaning no setter at all).

Comment (drwill-ms) [reply iothub/service/src/DigitalTwin/Models/DigitalTwinGetResponse.cs:22]: Like in `DigitalTwinUpdateResponse`

**Review by drwill-ms — APPROVED** [review]
Optimistic sign off.

**Pull request merged**